include/wx/charts/wxchartgrid.h

```diff
@@ -102,49 +102,67 @@ class wxChartGrid : public wxChartElement
     /// will be shown on the chart for the Y coordinate.
     /// @param options The settings to be used for the
     /// grid.
-	wxChartGrid(const wxPoint2DDouble &position, const wxSize &size,
-		wxDouble minXValue, wxDouble maxXValue,
-		wxDouble minYValue, wxDouble maxYValue,
-		const wxChartGridOptions& options);
+    wxChartGrid(const wxPoint2DDouble &position, const wxSize &size,
+        wxDouble minXValue, wxDouble maxXValue,
+        wxDouble minYValue, wxDouble maxYValue,
+        const wxChartGridOptions& options);
 
-	virtual bool HitTest(const wxPoint &point) const;
+    virtual bool HitTest(const wxPoint &point) const;
 
-	virtual wxPoint2DDouble GetTooltipPosition() const;
+    virtual wxPoint2DDouble GetTooltipPosition() const;
 
-	/// Draws the grid.
-	/// @param gc The graphics context.
-	void Draw(wxGraphicsContext &gc);
+    /// Draws the grid.
+    /// @param gc The graphics context.
+    void Draw(wxGraphicsContext &gc);
 
-	/// Resizes the grid.
-	/// @param size The new size of the area where the grid 
-	/// is displayed.
-	void Resize(const wxSize &size);
+    /// Resizes the grid.
+    /// @param size The new size of the area where the grid
+    /// is displayed.
+    void Resize(const wxSize &size);
 
-	/// Gets the mapping that allows external code
-	/// to correctly position points on the grid.
-	/// @return The grid mapping.
-	const wxChartGridMapping& GetMapping() const;
+    /// Gets the mapping that allows external code
+    /// to correctly position points on the grid.
+    /// @return The grid mapping.
+    const wxChartGridMapping& GetMapping() const;
+
+    bool Scale(int coeff);
+    void Shift(double dx,double dy);
 
 private:
-    static wxChartAxis::ptr CreateNumericalAxis(wxDouble minValue, wxDouble maxValue, 
+    static wxChartAxis::ptr CreateNumericalAxis(wxDouble minValue, wxDouble maxValue,
         const wxChartAxisOptions &options);
-	void Fit(wxGraphicsContext &gc);
-	void CalculatePadding(const wxChartAxis &xAxis,
-		const wxChartAxis &yAxis, wxDouble &left, wxDouble &right);
+    void Fit(wxGraphicsContext &gc);
+    void Update();
+    void CalculatePadding(const wxChartAxis &xAxis,
+        const wxChartAxis &yAxis, wxDouble &left, wxDouble &right);
     static void DrawHorizontalGridLines(const wxChartAxis &horizontalAxis, const wxChartAxis &verticalAxis,
         const wxChartGridLineOptions &options, wxGraphicsContext &gc);
     static void DrawVerticalGridLines(const wxChartAxis &horizontalAxis, const wxChartAxis &verticalAxis,
         const wxChartGridLineOptions &options, wxGraphicsContext &gc);
 
 private:
-	wxChartGridOptions m_options;
-	wxPoint2DDouble m_position;
-	wxChartAxis::ptr m_XAxis;
-	wxChartAxis::ptr m_YAxis;
-	wxChartGridMapping m_mapping;
-	// Whether something has changed and we
-	// need to rearrange the chart
-	bool m_needsFit;
+    struct Corners
```

Comment:
Exactly what need. I'll update it. What about name of variables 
```
 m_curcorners;// short form for current corners
 m_orcorners;// original
```
change to 
```
m_CurAxisLimits;
m_OrAxisLimits;
```
or better use something else?